spec/mongo/collection/view_spec.rb

```diff
@@ -545,11 +688,9 @@
             end
 
             it 'overrides the modifier value with the option value' do
-              expect(query_spec[:selector][:$query]).to eq(selector)
+              expect(query_spec[:selector][:$query]).to eq(options[:modifiers][:$query])
```

Comment:
In PyMongo we don't do anything to try to correct that mistake. Since the options in modifiers are added to the criteria after the criteria is changed to {'$query': {'a': 1}} the query ends up being {'$query': {'b': 2}}.